src/openqaoa-core/algorithms/workflow_properties.py

```diff
@@ -1,77 +1,106 @@
-#
-#   Unless required by applicable law or agreed to in writing, software
-#   distributed under the License is distributed on an "AS IS" BASIS,
-#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#   See the License for the specific language governing permissions and
-#   limitations under the License.
-
-from typing import List, Dict, Optional, Union
+from typing import List, Optional, Union
 import numpy as np
-import qiskit
 from scipy.optimize._minimize import MINIMIZE_METHODS
-import scipy
-import numpy
 
-from openqaoa.basebackend import QuantumCircuitBase
-from openqaoa.devices import SUPPORTED_LOCAL_SIMULATORS
-from .parameters import Parameters
-from scipy.optimize._minimize import MINIMIZE_METHODS
-from openqaoa.optimizers.training_vqa import CustomScipyGradientOptimizer, PennyLaneOptimizer
-
-
-
-ALLOWED_PARAM_TYPES = ['standard', 'standard_w_bias', 'extended', 'fourier',
-                       'fourier_extended', 'fourier_w_bias', 'annealing']
-ALLOWED_INIT_TYPES = ['rand', 'ramp', 'custom']
-ALLOWED_MIXERS = ['x', 'xy']
-ALLOWED_MINIMIZATION_METHODS = MINIMIZE_METHODS + CustomScipyGradientOptimizer.CUSTOM_GRADIENT_OPTIMIZERS + PennyLaneOptimizer.PENNYLANE_OPTIMIZERS
-
-ALLOWED_QVM_DEVICES = ['Aspen-11', 'Aspen-M-1']
-ALLOWED_QVM_DEVICES.extend(f'{n}q-qvm' for n in range(2, 80))
+from ..optimizers.training_vqa import CustomScipyGradientOptimizer, PennyLaneOptimizer
+from ..backends.devices_core import SUPPORTED_LOCAL_SIMULATORS
+from ..backends.basebackend import QuantumCircuitBase
+from ..utilities import convert2serialize
+
+
+ALLOWED_PARAM_TYPES = [
+    "standard",
+    "standard_w_bias",
+    "extended",
+    "fourier",
+    "fourier_extended",
+    "fourier_w_bias",
+    "annealing",
+]
+ALLOWED_INIT_TYPES = ["rand", "ramp", "custom"]
+ALLOWED_MIXERS = ["x", "xy"]
+
+ALLOWED_MINIMIZATION_METHODS = (
+    MINIMIZE_METHODS
+    + CustomScipyGradientOptimizer.CUSTOM_GRADIENT_OPTIMIZERS
+    + PennyLaneOptimizer.PENNYLANE_OPTIMIZERS
+)
+
+ALLOWED_QVM_DEVICES = ["Aspen-11", "Aspen-M-1"]
+ALLOWED_QVM_DEVICES.extend(f"{n}q-qvm" for n in range(2, 80))
 
 ALLOWED_LOCAL_SIMUALTORS = SUPPORTED_LOCAL_SIMULATORS
-ALLOWED_IMBQ_GLOBAL =  ['ibmq_qasm_simulator', 'ibmq_armonk', 'ibmq_santiago', 'ibmq_bogota', 'ibmq_lima', 'ibmq_belem', 'ibmq_quito', 'simulator_statevector', 'simulator_mps', 'simulator_extended_stabilizer', 'simulator_stabilizer', 'ibmq_manila']
-ALLOWED_DEVICES  = ALLOWED_LOCAL_SIMUALTORS + ALLOWED_QVM_DEVICES + ALLOWED_IMBQ_GLOBAL
-
-
-class CircuitProperties(Parameters):
+ALLOWED_IMBQ_GLOBAL = [
+    "ibmq_qasm_simulator",
+    "ibmq_armonk",
+    "ibmq_santiago",
+    "ibmq_bogota",
+    "ibmq_lima",
+    "ibmq_belem",
+    "ibmq_quito",
+    "simulator_statevector",
+    "simulator_mps",
+    "simulator_extended_stabilizer",
+    "simulator_stabilizer",
+    "ibmq_manila",
+]
+ALLOWED_DEVICES = ALLOWED_LOCAL_SIMUALTORS + ALLOWED_QVM_DEVICES + ALLOWED_IMBQ_GLOBAL
+
+
+class WorkflowProperties:
```

Comment:
why `WorkflowProperties`? in the asdict methods we have called them `parameters`. I think we should be consistent between how we call it for the dadatabase and the object  names.

Reply:
In that case, I recommend changing the asdict method. I don't want to overuse the word parameters

Reply:
@Q-lds what do you think? 
Do we change `asdict['data']['input_parameters']` to something like `asdict['data']['input_properties']`?

Reply:
Raised issue in #182 continue discussion there.